remote.rb

```diff
@@ -8,19 +8,75 @@
 
 page = Nokogiri::HTML(browser.html)
 
-data = []
+data_arr = []
+
+
+itter = 1
 
 page.css('.owwb-cs-slide-list-account-item').each do |i|
     account  = i.css('.contract-number').text
     balance  = i.css('.owwb-cs-slide-list-amount-value').text
     currency = i.css('.owwb-cs-slide-list-amount-currency').text
 
-data.push(  
-    name:     account,
-    balance:  balance,
-    currency: currency
-)  
-end  
+    pag = []
+    i.css('li').each{|li|  
+        li.children.children.children.each{|i| pag << i.text.strip.gsub(/\s{2,100}/, '')   
+        } 
+    }  
+    pag.select{|i| i =~ /[a-z, 1-8]/}
+    
+    nature   = pag[pag.index("Produsul")+1]
+    
+
+    adress   = 'file:///home/zik/ruby/trans_list_' + itter.to_s + '.html'
+
+    browser.goto(adress)
+    itter += 1
+    transactions_page = Nokogiri::HTML(browser.html)
+    
+    tr_arr = []
+
+    transactions_page.css('tbody').each{|tbody|
+        tbody.children.each{|tr|  
+            if tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s{1,6}/) then
+                tr_arr << tr
+            end  
+        }  
+    }
+
+    trans_hash = []
+
+    tr_arr.each{|element|
+        trans_date = element.children.children.children[0].text.strip.gsub(/\"/,'')
```

Comment:
there is no need to assign all these variables if you use them only once in code. you can save result direct in hash `key`